plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -545,16 +549,31 @@ public void deployModel(
                     log.debug("Model content matches original hash value, continue deploying");
                     Map<String, Object> params = ImmutableMap
                         .of(MODEL_ZIP_FILE, modelZipFile, MODEL_HELPER, modelHelper, ML_ENGINE, mlEngine);
-                    Predictable predictable = mlEngine.deploy(mlModel, params);
-                    try {
-                        modelCacheHelper.setPredictor(modelId, predictable);
-                        mlStats.getStat(MLNodeLevelStat.ML_NODE_TOTAL_MODEL_COUNT).increment();
-                        modelCacheHelper.setModelState(modelId, MLModelState.DEPLOYED);
-                        listener.onResponse("successful");
-                    } catch (Exception e) {
-                        log.error("Failed to add predictor to cache", e);
-                        predictable.close();
-                        listener.onFailure(e);
+                    if (mlModel.getAlgorithm() == FunctionName.METRICS_CORRELATION) {
```

Comment:
If we add more function name for `MLExecutable`, we need to change this line. How about create a new method to check if the function is MLExecutable in `FunctionName` , another easy way is making `MLExecutable` function name follow some pattern like `ML_EXECUTABLE_METRICS_CORRELATION`